hazelcast/src/main/java/com/hazelcast/multimap/impl/MultiMapService.java

```diff
@@ -428,4 +429,20 @@ public void dispatchEvent(EventData event, EntryListener listener) {
         }
         return multiMapStats;
     }
+
+    private Address getReplicaAddress(IPartition partition, int backupCount, int replicaIndex) {
+        Address replicaAddress = partition.getReplicaAddress(replicaIndex);
+        int tryCount = REPLICA_ADDRESS_TRY_COUNT;
+        int memberSize = nodeEngine.getClusterService().getSize();
+
+        while (memberSize > backupCount && replicaAddress == null && tryCount-- > 0) {
```

Comment:
Unrelated to your PR, but the check `memberSize > backupCount` could lead to unnecessary waiting. You can have more members than the configured backup count and still not have as many replicas as you have backup count. This is because of partition groups. For instance, you have 4 members in 2 partition groups. Even though the map is configured with 3 backups and there are 4 members, there will be only 1 backup replica.
That's why `IPartitionService#getMaxAllowedBackupCount` should be used. A similar use case is [here](https://github.com/hazelcast/hazelcast-enterprise/blob/a734e8cf920ecc0a925f34e9ffa6f878d1629a60/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/replication/AbstractWanPublisher.java#L244).